README.md

```diff
@@ -52,6 +52,16 @@ In case you use an exception reporting service (e.g. Bugsnag) you might want to
 define an error handler. See [Customizing the error
 handler](#customizing-the-error-handler) for more information.
 
+**ActiveJob Dependency**
+
+The Maintenance Tasks framework relies on ActiveJob behind the scenes to run
+Tasks. The default queuing backend for ActiveJob is
+[asynchronous](https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters/AsyncAdapter.html).
+It is **strongly recommended** to change this to a persistent backend so that
+Task progress is not lost during code or infrastructure changes. For more
+information on configuring a queuing backend, take a look at the [ActiveJob
+documentation](https://guides.rubyonrails.org/active_job_basics.html#setting-the-backend).
```

Comment:
I feel like it's missing something along the lines of "you may be already using Sidekiq but may not have set the queue adapter". Or maybe we could give them a thing to grep or a line to run in their console (i.e. `active_job.queue_adapter`).

But I may be wrong, telling them we use ActiveJob and pointing them to the docs on how to configure the queue adapter should be good!

Reply:
I think the `Setting the Backend` section in the docs is pretty straightforward in terms of telling users they need to set it as a config option. If folks are still getting confused about using Sidekiq but not using ActiveJob, we can add an extra note later, but I'd rather keep it lean for now!